src/test/java/com/salesforce/storm/spout/dynamic/consumer/PartitionDistributorTest.java

```diff
@@ -84,6 +91,12 @@ public void testCalculatePartitionAssignment(int totalConsumers, int consumerInd
             { 1, 0, new int[]{ 0, 1, 2 }, new int[]{ 0, 1, 2 } },
             // One instance, first instance, three partitions not in order, all partitions
             { 1, 0, new int[]{ 2, 0, 1 }, new int[]{ 0, 1, 2 } },
+            // Three instances, first instance, four partitions, first two partitions
+            { 3, 0, new int[]{ 0, 1, 2, 3 }, new int[]{ 0, 1 } },
+            // Three instances, second instance, four partitions, third partition
+            { 3, 1, new int[]{ 0, 1, 2, 3 }, new int[]{ 2 } },
```

Comment:
@c-mimms This should cover the 20/18/35 scenario, and the next one the 20/19/35 scenario.